### PR TITLE
On Travis/Linux Capybara fails to find an unused port, so retry

### PR DESCRIPTION
**Pull request by georgebrock**
This branch has been around for a while, and a client project I've been working on is depending on it in their Gemfile.

### CODE REVIEW

File: lib/capybara_discoball/runner.rb [hunk 1]
```diff
@@ -20,7 +20,21 @@ def boot
 
       private
 
-      def with_webrick_runner
+      def with_webrick_runner(&block)
+        retry_count = 3
```
Comment (JoelQ) [inline lib/capybara_discoball/runner.rb]: Should this be a constant?

File: lib/capybara_discoball/runner.rb [hunk 2]
```diff
@@ -20,7 +20,21 @@ def boot
...
+
+        begin
+          launch_webrick(&block)
+        rescue Errno::EADDRINUSE => e
+          raise if retry_count <= 0
+
+          retry_count -= 1
+          puts e.inspect
+          retry
+        end
```
Comment (JoelQ) [inline lib/capybara_discoball/runner.rb]: Most of this is about retrys, not webrick. Thoughts on extracting method `with_retries(&block)` ?

File: lib/capybara_discoball/runner.rb (hunk 1 above)
Comment (seanpdoyle) [inline lib/capybara_discoball/runner.rb]: What about an argument with a default?

File: lib/capybara_discoball/runner.rb [hunk 3]
```diff
@@ -20,7 +20,21 @@ def boot
...
+        retry_count = 3
+
+        begin
+          launch_webrick(&block)
+        rescue Errno::EADDRINUSE => e
+          raise if retry_count <= 0
+
+          retry_count -= 1
+          puts e.inspect
+          retry
```
Comment (seanpdoyle) [inline lib/capybara_discoball/runner.rb]: What do you think about rearranging this block to utilize an `if ... else` block?

File: lib/capybara_discoball/runner.rb [hunk 4]
```diff
@@ -20,7 +22,23 @@ def boot
...
 
-      def with_webrick_runner
+      def with_webrick_runner(&block)
+        with_retries(RETRY_COUNT) { launch_webrick(&block) }
+      end
+
+      def with_retries(retry_count)
+        yield
+      rescue Errno::EADDRINUSE => e
+        if retry_count > 0
```
Comment (georgebrock) [inline lib/capybara_discoball/runner.rb]: I think `> 0` better captures the intent here, Hound.

**Comment by dleve123** [comment]
excited to get this merged @georgebrock! this just came up again in some perf improvements that we're making to our build infrastructure

**Comment by odlp** [comment]
I've rebased this branch on `master` and added a spec for the retry behavior. Hope that's A-OK.

**Comment by odlp** [comment]
This is now available if one updates from `v0.0.x` -> `v0.1.0`:
https://rubygems.org/gems/capybara_discoball/versions/0.1.0

**Pull request merged**